doc/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,51 @@
+
+## Troubleshooting
+
+### Common Pitfalls
+#### Login profile files
+When the SSH slaves plugin connects to a slave, it does not run an interactive shell. 
+Instead it does the equivalent of your running `ssh slavehost command...` a few times, 
+eventually to run `ssh slavehost java -jar ...`. Exactly what happens on the slave as a result of this depends on the SSHD implementation, 
+but OpenSSH runs this with `bash -c command ...` (or whatever your login shell is.)
+
+This means some of your login profiles that set up your environment is not read by your shell. See [this post](http://stackoverflow.com/questions/216202/why-does-an-ssh-remote-command-get-fewer-environment-variables-then-when-run-man) for more details.
+
+If your login shell does not understand the command syntax used (e.g. the fish shell), use the advanced options **Prefix Start Slave Command** and **Suffix Start Slave Command** to wrap the slave command in e.g. `sh -c " and "`.
+
+Example: For loading `~/.bash_profile` on macOS Slaves set **Prefix Start Slave Command** to `source ~/.bash_profile && ` (intended whitespace at the end).
+Make sure to reconnect the slave after changing the slaves commands.
+
+#### Overall recommendations:
+* It is recommended to use JDK nearest and in the same major version of Jenkins instance and Agents
+* It is recommended to tune the TCP stack on of Jenkins instance and Agents
+    * On Linux http://tldp.org/HOWTO/TCP-Keepalive-HOWTO/usingkeepalive.html    
+    * On Windows https://blogs.technet.microsoft.com/nettracer/2010/06/03/things-that-you-may-want-to-know-about-tcp-keepalives/
+    * On Mac https://www.gnugk.org/keepalive.html
```

Comment:
Suggested simplifications:

* Use a JDK version in the same major version as the Jenkins instance and Agents and preferably a close minor version.
* Tune the TCP stack on of Jenkins instance and Agents
    * [Linux](http://tldp.org/HOWTO/TCP-Keepalive-HOWTO/usingkeepalive.html)
    * [Windows](https://blogs.technet.microsoft.com/nettracer/2010/06/03/things-that-you-may-want-to-know-about-tcp-keepalives/)
    * [Mac](https://www.gnugk.org/keepalive.html)